flash/data/process.py

```diff
@@ -101,6 +101,151 @@ class PreprocessState:
+            Example::
+
+                * Input: Receive a folder path:
+
+                * Action: Walk the folder path to find image paths and their associated labels.
+
+                * Output: Return a list of image paths and their associated labels.
+
+        - ``load_sample``: Function to load a sample from metadata sample.
+            Example::
+
+                * Input: Receive an image path and its label.
+
+                * Action: Load a PIL Image from received image_path.
+
+                * Output: Return the PIL Image and its label.
+
+        - ``pre_tensor_transform``: Performs transforms on a single data sample.
+            Example::
+
+                * Input: Receive a PIL Image and its label.
+
+                * Action: Rotate the PIL Image.
+
+                * Output: Return the rotated PIL image and its label.
+
+        - ``to_tensor_transform``: Converts a single data sample to a tensor / data structure containing tensors.
+            Example::
+
+                * Input: Receive the rotated PIL Image and its label.
+
+                * Action: Convert the rotated PIL Image to a tensor.
+
+                * Output: Return the tensored image and its label.
+
+        - ``post_tensor_transform``: Performs transform on a single tensor sample.
+            Example::
+
+                * Input: Receive the tensored image and its label.
+
+                * Action: Flip the tensored image randomly.
+
+                * Output: Return the tensored image and its label.
+
+        - ``per_batch_transform``: Performs transforms on a batch.
+            In this example, we decided not to override the hook.
+
+        - ``per_sample_transform_on_device``: Performs transform on a sample already on a ``GPU`` or ``TPU``.
+            Example::
+
+                * Input: Receive a tensored image on device and its label.
+
+                * Action: Apply random transforms.
+
+                * Output: Return an augmented tensored image on device and its label.
+
+        - ``collate``: Converts a sequence of data samples into a batch.
+            Example::
+
+                * Input: Receive a list of augmented tensored images and their respective labels.
+
+                * Action: Collate the list of images into batch.
+
+                * Output: Return a batch of images and their labels.
+
+        - ``per_batch_transform_on_device``: Performs transform on a batch already on ``GPU`` or ``TPU``.
+            Example::
+
+                * Input: Receive a batch of images and their labels.
+
+                * Action: Apply normalization on the batch by substracting the mean
+                    and dividing by the standard deviation from ImageNet.
+
+                * Output: Return a normalized augmented batch of images and their labels.
+
+    .. note::
+
+        By default, each hook will be no-op execpt the collate which is PyTorch default collate.
```

Comment:
make the collate a link to the pytorch method.

Reply:
Done !